### PR TITLE
Negate wasn't passing the exponent to the BigDecimal constructor.

### PR DESCRIPTION
**Pull request by AdamWhiteHat**
Added more tests. 100% of tests pass now.

### CODE REVIEW

**Comment by Protiguous** [comment]
Good catch! I'll add it into the experimental branch.

**Comment by Protiguous** [comment]
Ah, I see why my version wasn't throwing on the test. It was already passing it to the multiply function.
I'll have to test, but I think your version might be faster (less calculations) with the same result.

This is what I have atm,
`public static readonly BigDecimal MinusOne = new( -1 );`
`public static BigDecimal Negate( BigDecimal value ) => value * MinusOne;`

**Comment by AdamWhiteHat** [comment]
Yeah, probably.

BigInteger.Negate flips the sign of a private int called _sign (e.g. _sign = - _sign), and that, in turn, emits a single IL op code: neg.

Whereas BigInteger.Multiply will instantiate a new object and perform long-hand multiplication--I see no special case to short circuit multiplication by -1.

Check out my Nth root code I recently added after this commit. Its really elegant. According to [Wikipedia](https://en.wikipedia.org/wiki/Methods_of_computing_square_roots#Approximations_that_depend_on_the_floating_point_representation), Nth root for a floating point number is really simple: Find the Nth root of the mantissa, and divide the exponent by N.

I may add trigonometric function (sine, cosine, tangent) next/soon. Apparently the way to do trigonometric functions to arbitrary precision, digit by digit, is an algorithm called CORDIC. That will get you arbitrary precision tangent, then you can just convert the tangent to the sine/cosine for those. This works in/on radians only, but one can convert radian to/from degrees.
I have a proof of concept working for tangent on doubles, I just need to convert it to arbitrary precision and then write methods to go from tangent -> sine and tangent -> cosine.

I looked at other BigDecimal libraries available,  I don't think they even have Nth root available, definitely not trigonometric functions, so we are already ahead of them in terms of features.

File: BigDecimal/BigDecimal.cs [hunk 1]
```diff
@@ -744,7 +744,7 @@ namespace ExtendedNumerics;
 		left.Exponent > right.Exponent ? AlignExponent( left, right ) >= right.Mantissa : left.Mantissa >= AlignExponent( right, left );
 
 	/// <summary>Returns the result of multiplying a BigDecimal by negative one.</summary>
-	public static BigDecimal Negate( BigDecimal value ) => new BigDecimal( BigInteger.Negate( value.Mantissa ) );
+	public static BigDecimal Negate( BigDecimal value ) => new BigDecimal( BigInteger.Negate( value.Mantissa ), value.Exponent );
```
Comment (Protiguous) [inline BigDecimal/BigDecimal.cs:747]: Added in my experimental branch.

**Pull request merged**